lib/ui/painting/path_measure.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/path_measure.h"
+
+#include <math.h>
+
+#include "flutter/lib/ui/painting/matrix.h"
+#include "lib/tonic/converter/dart_converter.h"
+#include "lib/tonic/dart_args.h"
+#include "lib/tonic/dart_binding_macros.h"
+#include "lib/tonic/dart_library_natives.h"
+
+using tonic::ToDart;
+
+namespace blink {
+
+typedef CanvasPathMeasure PathMeasure;
+
+static void PathMeasure_constructor(Dart_NativeArguments args) {
+  DartCallConstructor(&CanvasPathMeasure::Create, args);
+}
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, PathMeasure);
+
+#define FOR_EACH_BINDING(V)          \
+  V(PathMeasure, setPath)            \
+  V(PathMeasure, getLength)          \
+  V(PathMeasure, getMatrix)          \
+  V(PathMeasure, getPosTan)          \
+  V(PathMeasure, getSegment)         \
+  V(PathMeasure, isClosed)           \
+  V(PathMeasure, nextContour)
+
+FOR_EACH_BINDING(DART_NATIVE_CALLBACK)
+
+void CanvasPathMeasure::RegisterNatives(tonic::DartLibraryNatives* natives) {
+  natives->Register({{"PathMeasure_constructor", PathMeasure_constructor, 3, true},
+                     FOR_EACH_BINDING(DART_REGISTER_NATIVE)});
+}
+
+fxl::RefPtr<CanvasPathMeasure> CanvasPathMeasure::Create(const CanvasPath* path, 
+                                                          bool forceClosed) {
+  fxl::RefPtr<CanvasPathMeasure> pathMeasure = fxl::MakeRefCounted<CanvasPathMeasure>();
```

Comment:
Any reason the ctor of `CanvasPathMeasure` cant just take the path instead of resorting to ivar access?

Reply:
Fair enough.